#! /bin/sh -e

## DP: Change default Debian value in accordance with Debian Policy
## DP: Author: Joel Johnson <mrjoel@lixil.net>

. $(dirname $0)/DPATCH

exit 0
@DPATCH@
diff -urN dovecot-1.2.1/dovecot-example.conf dovecot-1.2.1-debian/dovecot-example.conf
--- dovecot-1.2.1/dovecot-example.conf	2009-06-11 12:19:47.000000000 -0600
+++ dovecot-1.2.1-debian/dovecot-example.conf	2009-07-17 08:33:23.000000000 -0600
@@ -481,8 +481,14 @@
 # in is important to avoid deadlocks if other MTAs/MUAs are using multiple
 # locking methods as well. Some operating systems don't allow using some of
 # them simultaneously.
+#
+# The Debian value for mbox_write_locks differs from upstream Dovecot. It is
+# changed to be compliant with Debian Policy (section 11.6) for NFS safety.
+#       Dovecot: mbox_write_locks = dotlock fcntl
+#       Debian:  mbox_write_locks = fcntl dotlock
+#
 #mbox_read_locks = fcntl
-#mbox_write_locks = dotlock fcntl
+#mbox_write_locks = fcntl dotlock
 
 # Maximum time in seconds to wait for lock (all of them) before aborting.
 #mbox_lock_timeout = 300
diff -urN dovecot-1.2.1/src/master/master-settings.c dovecot-1.2.1-debian/src/master/master-settings.c
--- dovecot-1.2.1/src/master/master-settings.c	2009-06-30 09:02:57.000000000 -0600
+++ dovecot-1.2.1-debian/src/master/master-settings.c	2009-07-17 08:37:05.000000000 -0600
@@ -255,7 +255,7 @@
 	MEMBER(maildir_copy_preserve_filename) FALSE,
 	MEMBER(maildir_very_dirty_syncs) FALSE,
 	MEMBER(mbox_read_locks) "fcntl",
-	MEMBER(mbox_write_locks) "dotlock fcntl",
+	MEMBER(mbox_write_locks) "fcntl dotlock",
 	MEMBER(mbox_lock_timeout) 300,
 	MEMBER(mbox_dotlock_change_timeout) 120,
 	MEMBER(mbox_min_index_size) 0,
@@ -1822,7 +1822,15 @@
 	const char *set_names[4];
 	unsigned int count;
 
-	sets[0] = &default_settings;
+	/* Debian change - the Debian default for mbox_write_lock differs from
+		upstream Dovecot, however we still want 'dovecot -n' to report
+		based on Dovecot defaults to make it clear that the setting is
+		different, so we make a copy of the default_settings and force
+		it to report based on strict Dovecot defaults
+	*/
+	struct settings dovecot_default_settings = default_settings;
+	dovecot_default_settings.mbox_write_locks = "dotlock fcntl";
+	sets[0] = &dovecot_default_settings;
 	sets[1] = set->defaults;
 
 	set_names[0] = NULL;
